python/feathub/feature_views/derived_feature_view.py

```diff
@@ -98,16 +102,34 @@ def build(
                 feature_descriptors=[self.source], force_update=force_update
             )[0]
 
+        default_field_name_index = 0
+        while f"f{default_field_name_index}" in [
+            x.name for x in source.get_output_features()
+        ]:
+            default_field_name_index += 1
+
         features = []
         for feature in self.features:
             if isinstance(feature, str):
                 feature = self._get_feature_from_feature_str(
-                    feature, registry, source, force_update
+                    feature,
+                    registry,
+                    source,
+                    force_update,
+                    f"f{default_field_name_index}",
                 )
+                if feature.name == f"f{default_field_name_index}":
+                    default_field_name_index += 1
             features.append(feature)
 
         self._validate(features, source)
 
+        self._resolve_join_feature_dtype(
```

Comment:
It seems a bit weird to do `self._resolve_join_feature_dtype` after `self._validate(...)`.

I suppose we need this `_resolve_join_feature_dtype` only if user specifies a string-valued expression. But if that is the case, should dtype already be resolved by the _get_feature_from_feature_str() (which is invoked above)?